test/Microsoft.TestPlatform.Client.UnitTests/Execution/TestRunRequestTests.cs

```diff
@@ -140,6 +142,66 @@ public void CancelAsyncIfTestRunStateInProgressCallsExecutionManagerCancel()
             executionManager.Verify(dm => dm.Cancel(), Times.Once);
         }
 
+
+        [TestMethod]
+        public void OnTestSessionTimeoutShouldCallCancel()
+        {
+            this.testRunRequest.ExecuteAsync();
+            this.testRunRequest.OnTestSessionTimeout(null);
+            this.executionManager.Verify(o => o.Cancel(), Times.Once);
+        }
+
+        [TestMethod]
+        public void OnTestSessionTimeoutShouldLogMessage()
+        {
+            bool handleLogMessageCalled = false;
+            bool handleRawMessageCalled = false;
+
+            this.testRunRequest.TestRunMessage += (object sender, TestRunMessageEventArgs e) =>
+                {
+                    handleLogMessageCalled = true;
+                };
+
+            this.testRunRequest.OnRawMessageReceived += (object sender, string message) =>
+                {
+                    handleRawMessageCalled = true;
+                };
+
+            this.testRunRequest.OnTestSessionTimeout(null);
+
+            Assert.IsTrue(handleLogMessageCalled, "OnTestSessionTimeout should call HandleLogMessage");
+            Assert.IsTrue(handleRawMessageCalled, "OnTestSessionTimeout should call HandleRawMessage");
+        }
+
+        [TestMethod]
+        public void OnTestSessionTimeoutShouldGetCalledWhenExecutionCrossedTestSessionTimeout()
+        {
+            string settingsXml =
+                @"<?xml version=""1.0"" encoding=""utf-8""?>
+                <RunSettings>
+                     <RunConfiguration>
+                       <TestSessionTimeout>1000</TestSessionTimeout>
+                     </RunConfiguration>
+                </RunSettings>";
+
+            var testRunCriteria = new TestRunCriteria(new List<string> { "foo" }, 1, true, settingsXml);
+            var executionManager = new Mock<IProxyExecutionManager>();
+            var testRunRequest = new TestRunRequest(testRunCriteria, executionManager.Object);
+
+            ManualResetEvent onTestSessionTimeoutCalled = new ManualResetEvent(true);
+
+            executionManager.Setup(o => o.StartTestRun(testRunCriteria, testRunRequest)).Callback(() => { System.Threading.Thread.Sleep(3 * 1000); });
+
+            executionManager.Setup(o => o.Cancel()).Callback(() => onTestSessionTimeoutCalled.Set());
+
+            testRunRequest.ExecuteAsync();
+
+            onTestSessionTimeoutCalled.WaitOne(20 * 1000);
+
+            executionManager.Verify(o => o.Cancel(), Times.Once);
+
```

Comment:
nit: please remove extra lines. Divide every test into three blocks of code: arrange, act and assert.